src/Monolog/Processor/HostnameProcessor.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog\Processor;
+
+/**
+ * Adds value of gethostname into the log extra
+ *
+ * @author Billie Thompson
+ */
+class HostnameProcessor
+{
+    public function __invoke(array $record): array
+    {
+        $record['extra']['hostname'] = gethostname();
```

Comment:
I've added in caching, which could improve performance, the disadvantage now is that in long running processes, if the hostname changes while the process is running, we won't reflect that in the logs.